tasks/post-to-cmr/tests/cmr-test.js

```diff
@@ -95,35 +96,66 @@ test.serial('postToCMR throws error if CMR correctly identifies the xml as inval
 });
 
 test.serial('postToCMR succeeds with correct payload', async (t) => {
-  const newPayload = t.context.payload;
+  const { bucket, payload } = t.context;
+  const newPayload = payload;
   const granuleId = newPayload.input.granules[0].granuleId;
   const key = `${granuleId}.cmr.xml`;
+  newPayload.input.cmrRevisionId = 123;
+  newPayload.input.granules[1] = {
+    granuleId: 'L0A_RAD_RAW_product_0017-of-0020',
+    files: [
+      {
+        name: 'L0A_RAD_RAW_product_0017-of-0020.cmr.json',
+        bucket: `${t.context.bucket}`,
+        filename: `s3://${t.context.bucket}/L0A_RAD_RAW_product_0017-of-0020.cmr.json`,
+        type: 'metadata',
+        fileStagingDir: 'file-staging/subdir',
+        etag: '"13f2bb38e22496fe9d42e761c42a0e67"',
+      },
+    ],
+  };
```

Comment:
I'm trying to understand why we needed to add a second UMM-G granule to the input for this test? Were you trying to show that the revision ID gets used when posting either type of CMR metadata? 

In any case, I'm trying to understand how this test is proving the use of the revision ID for either data type. We've stubbed `ingestUMMGranule` and `ingest` to return the same revision ID as the test setup, so the fact that errors aren't thrown is what proves that it's working? 

Ultimately I think the error throwing is handled on the CMR side, so I think we need to prove is that the revision ID passed as input to this task is received by `ingest` and `ingestUMMGranule` as an argument. And then the unit tests for those functions prove whether it is used correctly.

I would remove the revision ID behavior from this test since it's supposed to be a verification that the basic case of publishing to CMR works. Revision ID is an optional and additional behavior, so I would just add two more small unit tests proving the argument is passed on to `ingest` and `ingestGranule`

Reply:
> I'm trying to understand why we needed to add a second UMM-G granule to the input for this test? Were you trying to show that the revision ID gets used when posting either type of CMR metadata?

Yes, exactly

Reply:
OK. Ultimately based on the understanding that CMR is what will throw the error, I think all we need to prove is that we've passed along the revision ID correctly to the code that publishes to CMR. 

So I would revert the changes to this test, which is just supposed to be a basic demonstration that the Lambda handler succeeds, and add one small unit test proving the argument is passed on to publish2Cmr.

Reply:
Hm, why is it wrong to update the test to account for ingesting a UMM-G granule?

Reply:
I guess it's somewhat philosophical. I think unit tests in general should only be testing one code path through your code, otherwise it's actually a unit test of multiple things, not just one. And the logic for updating UMM-G granules vs ECHO-10 granules are separate code paths, so if we're going to test both here, I'd say they should be separate unit tests.

But more fundamentally, `post-to-cmr` just calls `publish2Cmr`, which is where the logic for ECHO10 vs UMM-G is handled. So I'm not sure `post-to-cmr` itself needs to test the code path for both types of granules in the name of good test isolation. That seems like a job for tests of `publish2Cmr`.

So I'd be in favor of removing the UMM-G testing logic from this test and adding a separate test just to prove that the revision ID from the input to post-to-cmr is passed into publish2Cmr

Reply:
Ah, that is true. That makes sense to me. I've updated the unit tests so that `publish2CMR` checks for both `ECHO10` and `UMM-G`.